### ISSUE TITLE
Lesson categories suggestion

### ISSUE THREAD

**Issue by acrymble** [programminghistorian/jekyll#193]
Suggest new lesson categories to replace the current ones:
- Acquiring data
- Transforming data
- Analyzing Data
- Presenting Data
- Sustaining data

That way readers arriving at the lessons page can get a better mental picture of how the lessons fit with the various stages of the research workflow. Clearer than starting off with APIs, which most people won't know or why they'd want to learn it.

Related, @fredgibbs has suggested we go 1 lesson per row instead of 3 on the lessons page, with extra metadata shown (eg, lesson difficulty, abstract, date of publication, etc).

Please provide feedback.

**Comment by ahegel** [comment]
Trying to make a mock-up of the new page helped me figure out a few questions I have:
- By restructuring, we'll be breaking apart the current groupings of lessons by topic (e.g. lessons about Zotero) - should we provide ways of sorting the lessons into the old groupings, and/or by date, difficulty, etc.?
- Do all of the lessons neatly fit into these five categories? How will we make executive decisions on border cases?
- Is the focus on "Data" (appearing in every heading) what we want, or should it be more about projects, or the research process? I'm not sure myself.

In case you're a visual thinker, here's my [rough mock-up](https://dl.dropboxusercontent.com/u/58260988/lessonspage2.jpg).

**Comment by wcaleb** [comment]
I like the mockup, @ahegel! We may be able to make the top menu look cleaner by hiding some of that text with javascript until clicked on. 

I share the reservation about the categories being so "Data" centric. Not sure I have any really good alternative suggestions, though. "Source material" would be more recognizable to historians new to programming, but that might be too wordy. For some of the categories, data still makes sense (like "transforming," perhaps) but I agree with @ahegel that carrying it through all the categories might not be what we want.

**Comment by acrymble** [comment]
There's also the challenge of maintaining wide appeal. No matter what our
mission, most of our users are non-historians. I share the concern about
'data'.

On Tue, Feb 23, 2016 at 8:55 PM, W. Caleb McDaniel <notifications@github.com

**Comment by ianmilligan1** [comment]
If we were aiming solely at historians, I'd say "sources" - maybe "information" if more generally?

**Comment by acrymble** [comment]
Sources is the historian-ey way. Materials also might work. Or just 'digitally':

'Acquiring digitally', etc.

Maybe too abstract.

**Comment by wcaleb** [comment]
"Digital Materials"? "Information" might work, too, a la "Geographic Information Systems."

**Comment by ianmilligan1** [comment]
I kinda like "Digital Materials."

**Comment by acrymble** [comment]
maybe we should user test with our audience.

**Comment by fredgibbs** [comment]
I think this is a really important issue, actually--way more than semantics. I'd like to think about this more and put it on the agenda for next call. We could also just "gathering", "organizing", "analyzing", etc. as intransitives in order to foreground the emphasis on stage in the research process rather than "data" vs "sources" vs etc. Awesome work @ahegel!

**Comment by acrymble** [comment]
I should add, I love the mockup @ahegel I would definitely support that change.

**Comment by miriamposner** [comment]
Agreed!

**Comment by jerielizabeth** [comment]
I love the mockup as well @ahegel! The emphasis on "data" jumped out at me as well. I think it might also be useful to think about digital storytelling as a mode of being a "Programming Historian" that might not default so much to data (even though at some level where there is computation there is data.) I look forward to the conversation on the call!

**Comment by acrymble** [comment]
Just to add, we had an event last month at which a number of people with varying DH skills tried out and critiqued our site. This was one of the big areas they suggested for improvement, as APIs are the first thing you see and mean nothing to most people. One attendee suggested categories that end in 'ing' words as a practice rooted deeply in library science, reinforcing just going with:

```
Acquiring
Transforming
Analyzing
Presenting
Sustaining
```

**Comment by acrymble** [comment]
I mentioned to some of you that I had printed a copy of the PH. That involved playing around with a possible new ordering of lessons under the above headlines. You can download and look at my choices in this PDF: https://zenodo.org/record/49873#.VxMpnD-Ga7p

I may have put some in wrong containers, but let's talk about this at our next meeting.

**Comment by tapar001** [comment]
Love the mockup @ahegel! And I agree overall that leading with categories rather than APIs is a good move.

**Comment by acrymble** [comment]
We're going to look for someone to help us do this over the summer.

**Comment by amandavisconti** [comment]
I like the new categories and @ahegel's mockup! Would there be any worth in using [Isotope JS](http://isotope.metafizzy.co/) to let visitors find/sort lessons by a couple parameters: the new categories as default, but maybe also by tool (name: Omeka and/or method: topic modeling), by date, and by difficulty (as @ahegel suggests?). 

@acrymble Not sure if you'd found someone to help with this over the summer already, but if not I'd be happy to volunteer (I've used Isotope for a past project & liked it).

**Comment by acrymble** [comment]
Thanks @amandavisconti I've not used Isotope JS. @wcaleb is primarily responsible for keeping the website sustainable so I'll have to let him chime in on that. I've tried to get a start on this over on our submissions site: https://github.com/programminghistorian/ph-submissions/blob/gh-pages/lessons/index.md

But as you can see I didn't get very far, in part because of the way CSS is set up over there (see: https://github.com/programminghistorian/ph-submissions/issues/25).

I'd be quite happy to get some help with it, because I think this would make a big difference to useability.

**Comment by acrymble** [comment]
@amandavisconti, related to #180 
We have been added to the DOAJ repository, which is good because it makes it easier for institutions like libraries to index us. We've been asked to add the following however:

1) An annual publications page in addition to our /lessons directory. I suspect this should be quite easy to do?

Can this be part of your new lessons page redesign, or is that outside of the scope?

**Comment by amandavisconti** [comment]
Does "an annual publications page" mean a page dividing the lessons by year published? If so, I could either make that a filtering option (sort lessons by year published) and/or make a separate page (that pulls in new lessons, maybe by adding "year published" metadata to each lesson's yaml?)—preferences?

**Comment by acrymble** [comment]
Yes they just want to see what was published each year. They're still thinking in terms of the way old-style journals operate. The date of publication is already in each lesson's YAML.

A sort option should be fine if that's the technology you're already using.

**Comment by amandavisconti** [comment]
Great, I'll include this feature.

**Comment by amandavisconti** [comment]
Pre-team-call update! I haven't finished this task yet, but [the top three commits here](https://github.com/amandavisconti/jekyll/commits/improved-lessons-page) add and clean lesson front matter data, readying the lessons to be sorted in various ways on the lessons page and to make the lessons page less hard-coded (so new lessons are automatically pulled in and properly formatted, and each of the existing sections starting with the API section can be dynamically written from _data/topics.yml). 

I created several new lesson front-matter variables:
- activity (takes one of the five gerunds suggested above, e.g. acquiring)
- topic (can take multiple tags, e.g. Python, mapping)
- writing-language (so we can sort once we have some non-English lessons)
- abstract
- path (lesson-slug-name so it's easy to loop in both the lesson link and title image on the lessons page)

I added these to every lesson and also reorganized each lesson's front matter so that they each 
- include the full set of possible variables (some lessons were missing existing variables like difficulty)
- include all the variables in the same order (not sure that's super useful other than to let people skim the front matter, but it was easy to do while adding the new variables)

I tried to fill in as many variables as I could without reading the full lesson (e.g. every lesson now has an abstract). I'll share more details when I've finished setting up the lessons page to actually sort/filter these lessons.

**Comment by amandavisconti** [comment]
**Notes from the 9/29 call:**
- Address that current and new lesson YAML, as well as lessons/index.md, are in English (can we abstract so this stays synced? do we need a same-lesson-as variable?)
- Use least JS possible to add filter/sort functionality to lessons page

**Questions for followup after prototype ready/with team:**
- Which sort should be the default? (activity, publication date, something else?)
- Categories is doing something with permalinks, so I left those alone and created topics.yml separately (can I just ID what lessons have category in the path and add a redirect for those?)
- I only added topics to post headers when they were obvious (e.g. topics from the current lessons page section headers), since that requires knowledge of the full lesson. In the case of lesson with existing categories (original-ph, python), I did keep those as topics so they remain even if we address the category permalink thing and remove category as a variable.
- I created an abstract for every lesson by grabbing an early sentence or two. These can be improved.
- Need to update publication template & author instructions to fill in new header metadata correctly
- A couple things I'm not sure I can clean up from lesson headers: the (one?) "published: false" lesson; "published: true" (only one lesson?); a few lessons have "editors" metadata in addition to "reviewers" (if we want to keep this, can we add it for all the lessons?); "review-ticket" variable
- How to use lesson.previous/next to correctly order ordered lessons?

**Comment by acrymble** [comment]
Hey @amandavisconti, anything I can do to help at this stage?

**Comment by amandavisconti** [comment]
**Summary:** Rewrote lessons page to be less hard-coded, instead using liquid templating to loop through whatever we currently have in the /_lessons folder and the current YAML info for each lesson (so, don't move stuff to /_lessons unless it's ready to be public). Changes to site file structure and lesson YAML support this. [This code is not yet on the live site.](https://github.com/programminghistorian/jekyll/issues/193#issuecomment-275692867)

**Notes:**

1. Added metadata to lesson headers: abstract, activity (acquiring, transforming, etc.), topics, writing-language, slug (for link and title image use), difficulty (if lesson header doesn't contain already; should be 1-5, but adding "0" for unranked lessons for now so can call them as unsorted when sorting by difficulty), and reordered all lesson metadata so it uses the same template. 
2. Added topics YAML to all lessons. Current topics supported are in /_data/topics.yml.
3. For Jekyll to acknowledge collections (folders of files that you can call and loop through just like posts and pages), the folder in question needs to start with an underscore. I moved /lessons to /_lessons to allow this, so we should doublecheck that I've configured permalinks correctly such that the lesson URLs stay the same as before.
4. /deprecated was moved out of /_lessons and into /_oldsite, because absence of correct front matter in the deprecated files was breaking the site 
5. Added Open Journals requirement of lessons by publication date (default lesson sort is by date)
6. **Don't move anything into /_lessons unless it's ready to be public on the site**
7. Unpublished OCR lesson was moved to /deprecated (note there is a different, published OCR lesson left as is)
8. Inclusion of if previous/if next lesson loops was breaking lesson template; added to-do below to investigate and reimplement these

**Needed team input:**
1. Appearance (general look, functionality)? = positive on current appearance; get demo site up and test with focus groups
2. "[OCR with Tesseract and ScanTailor](http://programminghistorian.org/lessons/OCR-with-Tesseract-and-ScanTailor)" lesson is not appearing on lesson page => #318

**Comment by acrymble** [comment]
I don't recall exactly, but I suspect the deprecated lessons were put where they were for URL purposes? We're committed to not breaking links, especially because people use our lessons on syllabi. Does your plan affect that at all?

**Comment by amandavisconti** [comment]
@acrymble That makes sense! I'll put permalinks in those deprecated lessons' YAML so that the same URLs still point to them.

I'm still figuring out how to make my lesson work visible to the other team members (I don't want to mess with the /ph-submissions site if anyone's using it, and my work was forked from an older version of the live /jekyll site). I'm working on getting the site working via GitHub Pages + my own repo, but am still sorting out link structure. For now, [I also uploaded the files to Google Drive](https://drive.google.com/drive/folders/0Bxoh9EFX7i1HU1VDb3U4NHhKRW8?usp=sharing) so that those with Jekyll set up can also explore the site locally until I figure this out.

![screenshot_165](https://cloud.githubusercontent.com/assets/899033/22376395/c578a2d6-e47b-11e6-9701-7c55ccdf6484.png)
 Here's a screenshot of how the new page looks—I can do more CSS work (e.g. make it look like @ahegel's mockup) after feedback on today's call, if desired.

**Comment by ianmilligan1** [comment]
!!! I really wish I could make the call, but just wanted to say that this is fantastic looking!

**Comment by acrymble** [comment]
yeah that looks awesome! Can't wait to hear more.

**Comment by acrymble** [comment]
Looks great @amandavisconti.

Just aide memoire of what we discussed in the meeting:

1) Once we've got a live beta, we'd like to get some feedback from users to see if they have trouble finding what they're after, or otherwise have feedback.
2) We wondered out loud about whether or not we should adopt a DH ontology already published for this. @arojascastro suggested we look at TADIRAH: http://tadirah.dariah.eu/vocab/index.php. But for now, I think we should focus on getting a working version.

@wcaleb has volunteered to help as needed with the issues above.

**Comment by amandavisconti** [comment]
**Further work to do after demo up:**
- [x] Get code working on live site other team members can explore: [](https://amandavisconti.github.io/proghistdev/lessons/)
- [x] Incorporate feedback from team & focus group testing (Fred is handling styling/browser checks) ** Later comment has latest work on this, #333 has user testing commentary)**

**Comment by fredgibbs** [comment]
The preview looks great! I'm excited to see and fiddle with the beta version. I second the idea of gathering public feedback before we make an "official" switch. Also a good way to engage the community. 

One point that may be in a different thread (that I can't find right now) about styling the new lesson blocks, and also relates to a point above about Browser/OS/device/accessibility checks; I have been working on these issues simultaneously while modernizing the typography and adopting a more standard CSS framework. @amandavisconti, if you're super excited to do the lessons page typography/design yourself, i can stop; otherwise I'd love to do it while standardizing the rest of the site styles.

Regarding display order, would it be useful to have a YAML entry that sets lesson page display order? I'm not sure how entirely necessary it is at the moment, but I think we do have some (the API and GIS lessons), and will probably have more lessons that should be done in a certain order, and we should display them that way. Maybe this is already possible with the new template. I think we have a next lesson and previous lesson YAML variable for some lessons, but it might be worth standardizing that and using it for the lessons page as well.

**Comment by amandavisconti** [comment]
@fredgibbs Thanks! Your doing the typography/design would be awesome.

Order YAML would be great. Right now, the only links still hardcoded on the lessons page are the Python sequence, and it would be lovely to remove that. ([Related previous/next issue...](https://github.com/programminghistorian/jekyll/issues/321))

I've had the new lesson page code up on [another repo](https://github.com/amandavisconti/proghistdev) but need to fuss with the URLs to get the CSS/JS/gallery images showing up on the site there. Hoping to get that working on Monday and let the team know. [Here's the /lessons page cod](https://raw.githubusercontent.com/amandavisconti/proghistdev/master/lessons.md)e, if you're interested!

**Comment by fredgibbs** [comment]
@amandavisconti excellent! and thanks for the links!

**Comment by amandavisconti** [comment]
Demo page is now live at [https://amandavisconti.github.io/proghistdev/lessons/](https://amandavisconti.github.io/proghistdev/lessons/)

Please feel free to share with your students, etc. and share comments and ideas here!

Note that because of some URL structure stuff I haven't ironed out, some of the rest of that demo site (e.g. blog links are broken, GitHub logo isn't showing in footer, lesson citation URLs) is not working—but all the lesson categorization stuff is. We'll also be working on getting any lessons that should be displayed in a specific sequence (e.g. the Python lessons) doing that.

**Comment by acrymble** [comment]
Thanks @amandavisconti ! awesome stuff.

We might have to think about ordering. What gets to be at the top? Particularly important for new lessons so they get noticed. Reverse chronological order seems to be what you've got at the moment.

**Comment by ianmilligan1** [comment]
Wow that makes me feel old (the wget lesson is our oldest active one eh?).

I think newest at the top for the default 'all lessons' view would make the most sense? So my old trusty wget lesson would be at the bottom. 😄

**Comment by jerielizabeth** [comment]
@ianmilligan1 wget should always be first :) but I think reverse chronological makes the most sense.

**Comment by acrymble** [comment]
How should we solicit feedback on this? I might be able to get a few colleagues to kick the tires. But what would be useful to know @amandavisconti ?

**Comment by amandavisconti** [comment]
Asking people to look would be great! 

I'd like to know 
1. how people get to the lessons page (do they come for a specific lesson they heard about from someone? are they looking for a specific skill? or are they browsing to see if anything is exciting?)
2. are there other ways they'd like to sort/search/filter the lessons?
3. is this better than the old page, or more confusing?
4. is the new page harder to skim (we went from 3 lessons per line to 1)
In addition to general thoughts.

I don't remember if I have access to the @ProgHist Twitter account, but it would also be nice to tweet something like "We're revamping our lessons page to sort lessons by topic, difficulty, and writing language. What would be most useful to you?"

**Comment by acrymble** [comment]
Ok, I'll try to get a few colleagues roped in to test it out. It may take
me a few weeks.

**Comment by amandavisconti** [comment]
From user testing (#333) and previous tasks:
[x] Suggestions for tweaking language (remove date, author, reviewers and just leave lesson image, title, abstract)
[x] [alternative to reverse chronological ordering](https://github.com/programminghistorian/jekyll/issues/193#issuecomment-281208539)
[x] With the refactoring I did in 6e12c50, you should be able to access the language of the page with {{ page.lang }}. That variable is set by default in _config.yml based on the page's path. [*Updated #322 to use this.*]
[x] take out writing-language: in each lesson's YAML (#322 will address language sorting/filtering)
[x] Search box: indicate what this searches (_it searches lesson titles, abstracts, and authors_). Abstracts may be missing important topic words, so this search is probably not useful for most users. Let's get rid of it, but I'll write myself a note to discuss whole-site search with the team.

**Comment by fredgibbs** [comment]
i agree with basically everything in the previous post by @amandavisconti but i would say that a reverse chronological order is not necessarily that helpful. My sense is that many people attracted to PH need some hand-holding in terms of figuring out how to get started and learn totally new tech things, which we (thankfully) provide more than most tutorials. So I think there is a good argument to be made for the lesson to be organized by difficulty ahead of chronology within the larger categories. i can see the utility of a "newest lessons" button or something for regular visitors, but I think that is a much smaller audience than newcomers, or even people who have been here before but are trying to learn something new.

**Comment by amandavisconti** [comment]
## Done
- [x] Remove "original programming historian" topic from topic list and loop of lessons below it
- [x] Update publication template & author instructions to fill in new header metadata correctly (but note changed collections, writing language) [here](https://github.com/amandavisconti/proghistdev/blob/master/new-lesson-workflow.md#add-a-metadata-block)
- [x] Update publication template & author instructions to fill in new header metadata correctly [here](https://github.com/amandavisconti/proghistdev/blob/master/editor-guidelines.md). This is where the new lesson workflow page's "your editor will help you fill in the rest when your lesson is ready for publication" is satisfied.
- [x] check if writing-language, collections should still be in all lesson front matter and if it is in the actual lessons
- [x] Switched to reverse chronological lesson ordering for now, but the plan is to eventually fix lesson next/previous behavior and provide a path through the lessons on the lesson page (replacing reverse chrono order with a "newest lessons" sort if desired) via #321 
- [x] Test new local/demo baseurl stuff (can keep CNAME as it throws a warning, not a page-build-stopping error). Keeping the config with:
```
url: http://programminghistorian.org
baseurl: ""
```
and using `bundle exec jekyll serve --watch --baseurl '/proghistdev'`
lets me not switch config info between local/demo/PH sites (assuming we alter the main PH site config to include the empty baseurl line).
- [x] Some numerical indicator of how many lessons would be good ("7 of 56 match your criteria"). I added a live count of the number of lessons associated with a topic to the topic filter buttons.
- [x] Too much scrolling needed: removed 2nd display full list by topic and associated topic HTML anchors/links.
- [x] Implement filtering by topic. Changed activity filter to check for inclusion rather than ==, changed "web-scraping" topic to "web-scrape" as a hacky way to not deal with the inclusion of the letters "api" in "web-scraping". Currently hardcoded for all current topics, so will want to redo this as one iteration of the code that loops through latest count of topics from _data/topics.yml eventually.
- [x] Use difficulty metadata added in #320 to add sort by difficulty
- [x] Add sort by date (though note default sort is already by newest to oldest)

**Comment by amandavisconti** [comment]
[ ] Switched to reverse chronological lesson ordering for now, but the plan is to eventually fix lesson next/previous behavior and provide a path through the lessons on the lesson page (replacing reverse chrono order with a "newest lessons" sort if desired) via #321

**Comment by amandavisconti** [comment]
@wcaleb Your new templating work is great—just getting around to looking further at it today. I'm a bit confused about implementing [your suggestion](https://github.com/programminghistorian/jekyll/issues/335#issuecomment-282396830) that we may not need to rewrite `/lessons` as `/_lessons`, and wonder if you can help.

If I'm remembering correctly, I looked at implementing sorting/filtering `/lessons` and found that Jekyll wouldn't let you loop through and sort/filter a folder of files via its front matter (like you can with `/_posts` and `/_pages`) unless the folder was formatted as a Jekyll collection (`/_lessons`). The current demo lessons page uses that access to the front matter (e.g. `{{lesson.date}}`, `{{lesson.difficulty}}`).

That belief could be incorrect (I'm having trouble tracking down the documentation I thought supported this), and/or perhaps you're suggesting we make `/lessons` a subdirectory of `_pages` or `_posts` (avoiding creating a collection but still allowing us to sort/filter by lesson YAML), or something else?

**Comment by amandavisconti** [comment]
@wcaleb Realized `site` knows about `_posts`, `_pages`, and __categories_ (found [the SO answer I originally used](http://stackoverflow.com/a/9117390)). I had assumed /data/categories.yml meant Jekyll categories, but perhaps that's just a data file that happens to use the same word as its name?

**Comment by wcaleb** [comment]
@amandavisconti My thought was that you could set a "default" field-value for all of the pages that appear in the "lessons" directory in the `_config.yml` file. Something like this under the `defaults` line:

```
- 
    scope:
      path: "lessons"
    values:
      lesson: true
```

Then you could loop through all pages, checking with an if statement to see if `page.lesson` is true.

I'm not sure whether this is necessarily better or worse than collections, about which I know less. There may be slight performance differences? Or just less "hacky" code using yours instead of mine? Your solution with collections seems elegant to me, but I didn't know if collections is a more powerful solution than needed, or if it was responsible for some of the other problems (like the previous-next issue), so I was just throwing out this other possibility. Maybe @mdlincoln would want to weigh in with his thoughts.

P.S. Here's a page that discusses [use cases for collections](http://ben.balter.com/2015/02/20/jekyll-collections/), possibly of interest.

**Comment by amandavisconti** [comment]
## Lesson filter stuff that's ready:
Check out the demo at [https://amandavisconti.github.io/proghistdev/lessons](https://amandavisconti.github.io/proghistdev/lessons). Ignore the styling—Fred is working on that separately.
1. Filter by activity (live-updating count of lessons per activity)
2. Filter by topic (live-updating count of lessons containing that topic).
3. Sort lessons by difficulty or publication date (note default sort is by newest publication). This also works if you've applied one of the activity or topic filters above (e.g. sort to show easiest Omeka lesson).
4. Updated two places in documentation for [authors](https://github.com/amandavisconti/proghistdev/blob/master/new-lesson-workflow.md#add-a-metadata-block) and [editors](https://github.com/amandavisconti/proghistdev/blob/master/editor-guidelines.md) to understand the new lesson front matter structure and how to add new topics.

**Comment by fredgibbs** [comment]
i think the previous and next endless loop templating problem were caused because page.previous and page.next have special meaning for jekyll (at least in terms of collections). so we can use page.previous-lesson as the variable name or whatever else. This may no longer be an issue if we move stuff out of _lessons, but it might be a general feature of collections (in which case i think it will still happen). using a different YAML variable will resolve it in either case.

**Comment by amandavisconti** [comment]
## Tasks necessary before publication

- [x] Revert to /lessons and use new config's "defaults" (see [this](https://github.com/programminghistorian/jekyll/issues/193#issuecomment-284253305)). 
- [x] Get rid of front matter "collection" field.
- [x] Make /lessons accurately count lessons by topic now that collections are removed; replace site.lessons calls with something like {{ site.pages | lessons }}
- [x] Check that total and by-topic lesson counts are correct
- [x] Need to change documentation on 2 pages if removing "collection:" from front matter. [here](https://github.com/amandavisconti/proghistdev/edit/master/editor-guidelines.md) and [here](https://github.com/amandavisconti/proghistdev/blob/master/new-lesson-workflow.md#add-a-metadata-block)
- [x] Fork @fredgibbs repository, create a branch for your new lesson page
- [x] Incorporate your changes there
  - [x] Make sure to include changes to data/authors and includes/header
  - [x] Make sure to check header & footer links, blog posts, citation text and citation URL.
Kaleidoscope if git diffing too unpleasant.

- [x] Submit your changes to Fred's fork as a pull request
- [x] Ping @fredgibbs and @mdlincoln finish evaluating the merged changes there, before finally merging from his fork

**Comment by mdlincoln** [comment]
> Figure out what pieces of code need to be pushed to /programminghistorian/jekyll and which are remnants from the outdated clone I was working with.

@amandavisconti So your dev site is not a fork of the original codebase, but an entirely new git repo, correct? So we can't automatically merge in?

**Comment by amandavisconti** [comment]
@mdlincoln It's an entirely different repo, based on the code in the PH/jekyll repo from some point later summer last year. I forget why, but the non-forkage might have been my not knowing how else to host the page on GitHub Pages?

**Comment by mdlincoln** [comment]
OK! And yeah, that makes sense. It will have to be your own manual merge then. 

Given the workflow that I suggested in #414, what do you think of this merge process:
1. Fork @fredgibbs repository, create a branch for your new lesson page, and incorporate your changes there
2. Submit your changes to Fred's fork as a pull request
3. We finish evaluating the merged changes there, before finally merging from his fork (since it is a branch of the master PH repo and can be done automatically)?

**Comment by amandavisconti** [comment]
@mdlincoln Sounds perfect! I'm just shifting my code from using collections now, and that'll be my next step. Will ping when accomplished.

**Comment by mdlincoln** [comment]
👍 (I'll make a note of this in #414, too)
